src/config/cors_params.rs

```diff
@@ -0,0 +1,95 @@
+use hyper::header;
+use hyper::http::HeaderValue;
+use serde::{Deserialize, Serialize};
+use serde_unit_struct::{Deserialize_unit_struct, Serialize_unit_struct};
+
+#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq, Eq, schemars::JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct CorsParams {
+    #[serde(default)]
+    pub allow_credentials: bool,
+    #[serde(default)]
+    pub allow_headers: ConstOrMirror,
+    #[serde(default)]
+    pub allow_methods: ConstOrMirror,
+    #[serde(default)]
+    pub allow_origin: StringOrSequence,
+    #[serde(default)]
+    pub allow_private_network: bool,
+    #[serde(default)]
+    pub expose_headers: StringOrSequence,
+    #[serde(default)]
+    pub max_age: Option<usize>,
+    #[serde(default = "preflight_request_headers")]
+    pub vary: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, schemars::JsonSchema)]
+#[serde(untagged)]
+pub enum ConstOrMirror {
```

Comment:
Mirror might not be required at this moment. WE can simulate similar behaviour using `*`